File: server/plugin/geoMonitor/router/playwright_session.go
package router

import "github.com/gin-gonic/gin"

type playwrightSession struct{}

func (r *playwrightSession) Init(public *gin.RouterGroup, private *gin.RouterGroup) {
	group := private.Group("geoMonitor")
	group.POST("playwright/session/start", apiPlaywrightSession.Start)
	group.GET("playwright/session/:id", apiPlaywrightSession.Get)
	group.POST("playwright/session/:id/refresh", apiPlaywrightSession.Refresh)
	group.DELETE("playwright/session/:id", apiPlaywrightSession.Delete)
}

File: web/src/plugin/geoMonitor/api/playwrightSession.js
import service from '@/utils/request'

export const startPlaywrightSession = (data) => {
  return service({
    url: '/geoMonitor/playwright/session/start',
    method: 'post',
    data
  })
}

export const getPlaywrightSession = (id) => {
  return service({
    url: `/geoMonitor/playwright/session/${id}`,
    method: 'get'
  })
}

export const refreshPlaywrightSession = (id) => {
  return service({
    url: `/geoMonitor/playwright/session/${id}/refresh`,
    method: 'post'
  })
}

export const deletePlaywrightSession = (id) => {
  return service({
    url: `/geoMonitor/playwright/session/${id}`,
    method: 'delete'
  })
}
